ports/stm/boards/stm32f746g_disco/board.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Scott Shawcroft for Adafruit Industries
+ * Copyright (c) 2020 Mark Olsson <mark@markolsson.se>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "boards/board.h"
+#include "stm32f7xx_hal.h"
+#include "common-hal/microcontroller/Pin.h"
+
+void board_init(void) {
+    GPIO_InitTypeDef GPIO_InitStructure;
+
+    __HAL_RCC_GPIOK_CLK_ENABLE();
```

Comment:
This comes after `port_init` in `main()`, so the clocks should already be on.

Reply:
Once the clocks are dynamic to save power they'll get turned on with a `claim_pin()` instead.

Reply:
I think it's good practice to turn on the clocks in case something else turns them off...in this case we know the flow so that's not going to happen...but for safety's sake it could be good to keep it?

Reply:
Sure. The eventual intention is to have clocks turned on and off when a pin is reserved with `claim_pin`, which will have the effect you're thinking of, so we can just leave it until that gets in. I've been putting it off since Scott's close to finishing his low power code and I'd like to do it as a part of my port cleanup after that.